src/components/ProjectSettingsTabs/GithubCommitQueue/GithubCommitQueueTab.tsx

```diff
@@ -0,0 +1,117 @@
+import { useMemo } from "react";
+import Banner from "@leafygreen-ui/banner";
+import { SpruceForm, SpruceFormProps } from "components/SpruceForm";
+import { ProjectSettingsTabRoutes } from "constants/routes";
+import { environmentalVariables } from "utils";
+import { usePopulateForm, useProjectSettingsContext } from "../Context";
+import { getFormSchema } from "./getFormSchema";
+import { mergeProjectRepo } from "./transformers";
+import { AliasType, FormState, TabProps } from "./types";
+
+const { isProduction } = environmentalVariables;
+
+const tab = ProjectSettingsTabRoutes.GithubCommitQueue;
+
+const getInitialFormState = (
+  projectData: FormState,
+  repoData: FormState
+): FormState => {
+  if (!projectData) return repoData;
+  if (repoData) {
+    return mergeProjectRepo(projectData, repoData);
+  }
+  return projectData;
+};
+
+export const GithubCommitQueueTab: React.FC<TabProps> = ({
+  gitHubWebhooksEnabled,
+  projectData,
+  repoData,
+  useRepoSettings,
+}) => {
+  const { getTab, updateForm } = useProjectSettingsContext();
+  const { formData } = getTab(tab);
+
+  const initialFormState = useMemo(
+    () => getInitialFormState(projectData, repoData),
+    [projectData, repoData]
+  );
+  usePopulateForm(initialFormState, tab);
+
+  const onChange = updateForm(tab);
+
+  const { fields, schema, uiSchema } = useMemo(
+    () =>
+      getFormSchema(
+        useRepoSettings,
+        gitHubWebhooksEnabled,
+        formData,
+        useRepoSettings ? repoData : null
+      ),
+    [formData, gitHubWebhooksEnabled, repoData, useRepoSettings]
+  );
+
+  if (!formData) return null;
+
+  return (
+    <>
+      {!gitHubWebhooksEnabled && (
+        <Banner data-cy="disabled-webhook-banner" variant="warning">
```

Comment:
![image](https://user-images.githubusercontent.com/26798134/151845141-82a06eca-5977-4710-afda-8a7b80d58975.png)

FYI this doesn't prevent the user from saving the page with Enabled set for these things, and it should be, but I'm not sure if that's what line 71 is about.

Reply:
Yeah, the entire page will be disabled if GitHub webhooks are not enabled once we fix the EVG-16208 bug. For the meantime I'm preventing disabling so that the page can be tested.

Reply:
Oh right as we literally discussed 🤦‍♀️